### PR TITLE
I2c master api

### PR DESCRIPTION
**Pull request by wajahat-abbas**
Notes:
- Pull requests will not be accepted until the submitter has agreed to the [contributer agreement](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md).
- This is just a template, so feel free to use/remove the unnecessary things

## Description
Not to be merged yet. I2C master write and read is working however after a successful read transaction, peripheral is not releasing the SCL line. Need to look further into it

## Status

**IN DEVELOPMENT**

## Migrations

If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

YES | NO

## Related PRs

List related PRs against other branches:

branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()

## Todos

- [ ] Tests
- [ ] Documentation

## Deploy notes

Notes regarding the deployment of this PR. These should note any required changes in the build environment, tools, compilers and so on.

## Steps to test or reproduce

Outline the steps to test or reproduce the PR here.

### CODE REVIEW

**Comment by wajahat-abbas** [comment]
Not to be merged yet. I2C master write and read is working however after a successful read transaction, peripheral is not releasing the SCL line. Need to look further into it

File: targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h [hunk 1]
```diff
@@ -55,8 +55,17 @@ typedef enum {
     UART1_CTS = p12,
     UART1_RTS = p11,
 
-    I2C_SCL = p0,
-    I2C_SDA = p1,
+    I2C1_SCL = p0,
+    I2C1_SDA = p1,
+
+	I2C_SCL = I2C1_SCL,
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h]: Please replace tabs with spaces.

File: targets/TARGET_ublox/TARGET_R5030/i2c_api.c [hunk 2]
```diff
@@ -0,0 +1,656 @@
...
+    /* FIXME: reset interface after a frequency change? */
+
+    /* hz should not be zero */
+    MBED_ASSERT(hz);
+    hz=hz/1000; /* convert to KHz */
+    bool fast_mode;
+    unsigned char fr_value;  
+    uint16_t cc_value;  
+    uint32_t register_value;
+    uint32_t engine_frequency_in_KHz;
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/i2c_api.c]: Use all lower case to stick with the convention.

File: targets/TARGET_ublox/TARGET_R5030/i2c_api.c [hunk 3]
```diff
@@ -0,0 +1,656 @@
...
+}
+
+int i2c_byte_write(i2c_t *obj, int data)
+{
+    int ret_val;
+    uint32_t status;
+    uint32_t time_out=MAX_TIMEOUT;
+
+    /* wait until there is space in fifo */
+    while (!DRIVER_BITFIELD_GET(obj->reg_base->sr, I2C_SR_TXHALFFULL)); /* 0 = TX buffer contains more than 8 words, 1 = TX buffer contains less than or exact 8 words */
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/i2c_api.c]: Is there any danger of this locking up?  Could consider using a time-out here.

**Review by RobMeades — CHANGES_REQUESTED** [review]
A few very minor comments below.  One general thing: we must start following the mbed-os coding standard more closely.  It is described here: https://os.mbed.com/teams/SDK-Development/wiki/mbed-sdk-coding-style

**Comment by wajahat-abbas** [comment]
Will make a new branch

Edit: rebased it

File: targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h (hunk 1 above)
Comment (wajahat-abbas) [reply targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h]: I have made changes in commit "code clean-up"

**Pull request merged**